Makefile

```
@@ -1,77 +1,94 @@
-.PHONY: all osxkeychain secretservice test lint validate-vendor fmt validate wincred pass deb vendor
+PKG := github.com/docker/docker-credential-helpers
+VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
 
-VERSION := $(shell grep 'const Version' credentials/version.go | awk -F'"' '{ print $$2 }')
+GO_LDFLAGS=-s -w -X ${PKG}/credentials.Version=${VERSION} -X ${PKG}/credentials.Revision=${REVISION}
 
-all: test
+BUILDX_CMD ?= docker buildx
+DESTDIR ?= ./bin/build
 
+.PHONY: all
+all: cross
+
+.PHONY: clean
 clean:
 	rm -rf bin
-	rm -rf release
-
-osxkeychain:
-	mkdir -p bin
-	go build -ldflags -s -o bin/docker-credential-osxkeychain ./osxkeychain/cmd/
-
-osxcodesign: osxkeychain
-	$(eval SIGNINGHASH = $(shell security find-identity -v -p codesigning | grep "Developer ID Application: Docker Inc" | cut -d ' ' -f 4))
-	xcrun -log codesign -s $(SIGNINGHASH) --force --verbose bin/docker-credential-osxkeychain
-	xcrun codesign --verify --deep --strict --verbose=2 --display bin/docker-credential-osxkeychain
 
-secretservice:
-	mkdir -p bin
-	go build -o bin/docker-credential-secretservice ./secretservice/cmd/
+.PHONY: build-%
+build-%: # build, can be one of build-osxkeychain build-pass build-secretservice build-wincred
+	$(eval BINNAME ?= docker-credential-$*)
+	go build -trimpath -ldflags="$(GO_LDFLAGS) -X ${PKG}/credentials.Package=${PKG}/$* -X ${PKG}/credentials.Name=docker-credential-$*" -o $(DESTDIR)/$(BINNAME) ./$*/cmd/
```

Did you update, because this looks to still have `PKG` used both at the start and as value;

```suggestion
	go build -trimpath -ldflags="$(GO_LDFLAGS) -X ${PKG}/credentials.Package=${PACKAGE_NAME}/$* -X ${PKG}/credentials.Name=docker-credential-$*" -o $(DESTDIR)/$(BINNAME) ./$*/cmd/
```

(mostly thinking that `Package` here could, e.g., be `docker-ce` or some other name.

---

Oh I get it now, so for me `Package` is Go module or base package. Will change that with a new var like you said..

---

Yes, so wondering if `PKG` needs to be configurable in this context 🤔 (could just be hard-coded)

---

Yes hardcoded is fine, changed to `GO_PKG`